Readme.md

```diff
@@ -225,38 +263,175 @@ function BankAccount() {
 ```js
 function bank_Account() {
 }
+
+/** @constructor */
+var client = function() {
+};
+
+/** @enum {string} */
+var countryLocale = {
+  CHINA: 'zh-CN',
+  US: 'en-US'
+};
 ```
 
-## Use UPPERCASE for Constants
+### Use UPPERCASE for Constants
 
 Constants should be declared as regular variables or static class properties,
 using all uppercase letters.
 
-Node.js / V8 actually supports mozilla's [const][const] extension, but
-unfortunately that cannot be applied to class members, nor is it part of any
-ECMA standard.
+*NOTE*: Use of ES6 keyword `const` is okay in NodeJS / V8 engines but
+discouraged until ES6 becomes more widely accepted. And, variablenames should
+still be be UPPERCASE_LIKE_THIS.
+
+Annotation with `@const` is recommended for all constant variables.
 
 *Right:*
 
 ```js
+/** @const */
 var SECOND = 1 * 1000;
 
+var MINUTE = 60 * SECOND; // Okay, but discouraged.
+
 function File() {
 }
+
+/** @const */
 File.FULL_PERMISSIONS = 0777;
 ```
 
 *Wrong:*
 
 ```js
-const SECOND = 1 * 1000;
+var second = 1 * 1000;
 
 function File() {
 }
+
 File.fullPermissions = 0777;
 ```
 
-[const]: https://developer.mozilla.org/en/JavaScript/Reference/Statements/const
+### File and Directories
+
+File and directory names should be lower-lisp-case, such as `a-awesome-dir`,
+`a-super-cool-file.js`, or `some-random-class.js`.
+
+For test files, a suffix of `_test` must be inserted right before `.js` file
+extension; e.g. `a-super-cool-file_test.js`, or `some-random-class_test.js`.
```

Comment:
This was a custom cooked rule by me, I would actually be in favor of using `-` all over the file names as they are a bit easier to type. I'd be cool if we ended up picking one over another and stick with just one.